WinEventParser/main.py

```diff
@@ -20,38 +20,38 @@ def admin():
         return False
 
 
+def is_valid_char(char):
+    """Function to check if each char of timestamp are valid"""
+    acceptable_chars = ['-', ' ', ':']
+    return char.isdigit() or char in acceptable_chars
+
+
+def get_chars_to_add(mode):
+    """Function used to return the string to add to our timestamp"""
+    chars_to_add = ""
+    if mode == 2:
+        chars_to_add = ":00"
+    elif mode == 3:
+        chars_to_add = ":00:00"
+    elif mode == 4:
+        chars_to_add = " 00:00:00"
+    elif mode == 5:
+        chars_to_add = "-01 00:00:00"
+    elif mode == 6:
+        chars_to_add = "-01-01 00:00:00"
+    return chars_to_add
+
+
 def parse_timestamp(timestamp_to_be_parsed, mode):
     """Function used to parse the input timestamp we had"""
     if timestamp_to_be_parsed is None:
         return None, None
-    timestamp_to_be_parsed = timestamp_to_be_parsed.split()
-    if len(timestamp_to_be_parsed) == 2:
-        timestamp_to_be_parsed = timestamp_to_be_parsed[0] + ' ' + timestamp_to_be_parsed[1]
-    else:
-        timestamp_to_be_parsed = timestamp_to_be_parsed[0]
-    acceptable_chars = ['-', ' ', ':']
-    for ttbp in timestamp_to_be_parsed:
-        if (ttbp.isdigit() or ttbp in acceptable_chars) is not True:
-            return None, None
+    timestamp_to_be_parsed = ' '.join(timestamp_to_be_parsed.split()[:2])
```

Comment:
Line 49 through 51 should be shifted to a new function as it has markedly different functionality than the rest of the function.
Line 47 and 48 should likely move to this new function as well.